cpp/src/text/jaccard.cu

```diff
@@ -0,0 +1,305 @@
+ * of the corresponding input string row.
+ *
+ * The hashes are then sorted using a segmented-sort as setup to
+ * perform the unique and intersect operations.
+ */
+std::unique_ptr<cudf::column> hash_substrings(cudf::strings_column_view const& col,
+                                              cudf::size_type width,
+                                              rmm::cuda_stream_view stream)
+{
+  auto hashes = hash_character_ngrams(col, width, stream, rmm::mr::get_current_device_resource());
+  auto const input   = cudf::lists_column_view(hashes->view());
+  auto const offsets = input.offsets_begin();
+  auto const data    = input.child().data<uint32_t>();
+
+  rmm::device_uvector<uint32_t> sorted(input.child().size(), stream);
+
+  // this is wicked fast and much faster than using cudf::lists::detail::sort_list
+  rmm::device_buffer d_temp_storage;
+  size_t temp_storage_bytes = 0;
+  cub::DeviceSegmentedSort::SortKeys(d_temp_storage.data(),
+                                     temp_storage_bytes,
+                                     data,
+                                     sorted.data(),
+                                     sorted.size(),
+                                     input.size(),
+                                     offsets,
+                                     offsets + 1,
+                                     stream.value());
+  d_temp_storage = rmm::device_buffer{temp_storage_bytes, stream};
+  cub::DeviceSegmentedSort::SortKeys(d_temp_storage.data(),
+                                     temp_storage_bytes,
+                                     data,
+                                     sorted.data(),
+                                     sorted.size(),
+                                     input.size(),
+                                     offsets,
+                                     offsets + 1,
+                                     stream.value());
+
+  auto contents = hashes->release();
+  return cudf::make_lists_column(
+    col.size(),
+    std::move(contents.children.front()),
+    std::make_unique<cudf::column>(std::move(sorted), rmm::device_buffer{}, 0),
+    0,
+    rmm::device_buffer{},
+    stream,
+    rmm::mr::get_current_device_resource());
+}
+}  // namespace
+
+std::unique_ptr<cudf::column> jaccard_index(cudf::strings_column_view const& input1,
+                                            cudf::strings_column_view const& input2,
+                                            cudf::size_type width,
+                                            rmm::cuda_stream_view stream,
+                                            rmm::mr::device_memory_resource* mr)
+{
+  CUDF_EXPECTS(
+    input1.size() == input2.size(), "input columns must be the same size", std::invalid_argument);
+  CUDF_EXPECTS(width >= 2,
+               "Parameter width should be an integer value of 2 or greater",
+               std::invalid_argument);
+
+  auto constexpr output_type = cudf::data_type{cudf::type_id::FLOAT32};
```

Comment:
Found it, [here's an example](https://godbolt.org/z/o4YaEhrsE) (admittedly not immediately relevant here since it's for pointers, but it's a good reason to maintain a consistent rule that has the expected behavior everywhere).

Reply:
Wow. That just looks strange to me.